source/controllers/disk.py

```diff
@@ -152,8 +156,21 @@ def sync_disks():
                                                                     'filesystem': fs_type if fs_type != '' else None,
                                                                     'mountpoint': mount_point})
 
+        # Check names to avoid a unique constraint exception
+        for disk_name, disk_info in configuration.iteritems():
+            if len(disk_info['aliases']) >= 1:
+                try:
+                    disk = DiskList.get_by_alias(disk_info['aliases'][0])
+                    if disk_name != disk.name:
+                        DiskController._logger.info('Detected moved disks, changing disk names to circumvent unique constraints')
```

Comment:
'Detected moved disks' is a little unclear
I'd change it to: 'Disk with alias {0} its name has changed from {1} to {2}, changing ...